forwards_paper.tex

```diff
@@ -496,24 +497,20 @@ \subsection*{Tree sequence simplification}
 
 More concretely,
 the algorithm works by moving back through time,
-processing each edge in the input tree sequence in chronological order.
-The main state of the algorithm at each point in time is a set of ancestral segments:
-each segment $(\ell, r, u)$ represents a tract of ancestral material in which the node $u$ is
-associated with the genomic interval $[\ell, r)$. 
-These segments are stored in a collection of linked lists,
-one list of segments for each ancestral lineage present at that time.
+processing each parent in the input tree sequence in chronological order.
+The main state of the algorithm at each point in time is a set of ancestral lineages,
+and each lineage is a collection of ancestral segments.
+An ancestral segment $(\ell, r, u)$ is found in a lineage
+if the output node $u$ inherits the genomic interval $[\ell, r)$ from that lineage.
+% These segments are stored in a collection of linked lists,
+% one list of segments for each ancestral lineage present at that time.
 (These lineages are the ``colors'' above.)
-The node $u$ in each ancestral sequence
-is the node ID associated with this segment in the output tree sequence.
-We maintain a map $A$ such that $A_j$ is segment chain for node $j$ in
+We also maintain a map $A$ such that $A_j$ is segment chain for node $j$ in
 the input tree sequence.
+Crucially, the time required to run the algorithm is linear in the number of edges of the input tree sequence..
 
-\jk{I'm not sure this is the right level of detail to tackle this description
-at. What's the goal of this algorithm description? Are we trying to
-establish some properties of the algorithm or just give the reader a high-level
-flavour of how it works??}
-
-In this scheme, at any point in the simulation genealogical history is recorded
+\paragraph{Sequential simplification}
```

Comment:
This seems like the most useful place to do some calculations.

Reply:
actually, I think optimal simpliy interval is "wait as long as RAM allows you" - doing it more often just means chasing more colors up the tree sequence more times.  do you agree, @jeromekelleher ?

Reply:
Yes, I think this is probably true. It's a straightforward time-space tradeoff I think.